### PR TITLE
fix(sql): allow quotes around dedup keys

### PR DESCRIPTION
**Pull request by marregui**
closes https://github.com/questdb/questdb/issues/3688

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|1|1|100.00%|

File: core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java [hunk 1]
```diff
@@ -162,6 +162,28 @@ public void testCreateTableSetTypeSqlSyntaxErrors() throws Exception {
         });
     }
 
+    @Test
+    public void testCreateTableWithDoubleQuotes() throws Exception {
+        String tableName = "ABB_Events2";
+        assertMemoryLeak(ff, () -> {
+            ddl(
```
Comment (bziobrowski) [inline core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java:169]: Ugh! String literal  used as table name :)

File: core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java [hunk 2]
```diff
@@ -162,6 +162,28 @@ public void testCreateTableSetTypeSqlSyntaxErrors() throws Exception {
...
+                    "CREATE TABLE '" + tableName + "' (\n" +
+                            "  Status SYMBOL capacity 16 CACHE,\n" +
+                            "  Reportedtime TIMESTAMP\n" +
```
Comment (bziobrowski) [inline core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java:172]: Could you also test with quoted identifier that contains a whitespace character ?

Comment (marregui) [reply core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java]: 4175bcf16000896346840a4bc5907d020c44a1e7

File: core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java [hunk 3]
```diff
@@ -162,6 +162,28 @@ public void testCreateTableSetTypeSqlSyntaxErrors() throws Exception {
...
+            }
+            assertSql(
+                    "column\ttype\tindexed\tindexBlockCapacity\tsymbolCached\tsymbolCapacity\tdesignated\tupsertKey\n" +
+                            "Status\tSYMBOL\tfalse\t256\ttrue\t16\tfalse\tfalse\n" +
+                            "Reportedtime\tTIMESTAMP\tfalse\t0\tfalse\t0\ttrue\ttrue\n",
+                    "SHOW COLUMNS FROM " + tableName
+            );
+        });
+    }
+
```
[the remark this answers is not in the document]
Comment (marregui) [reply core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java:195]: 89c7a3a59e908d7b0829785f4b48e8789da5ea3e

File: core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java (hunk 1 above)
Comment (bziobrowski) [reply core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java:169]: I meant the single quotes  around table name in sql command. It should be double quotes instead .

Comment (marregui) [reply core/src/test/java/io/questdb/test/griffin/CreateTableDedupTest.java:169]: ah!, nope, we do not make distinction, see isQuote method. I would not want to be pedantic in this specific meaningless "issue" when the codebase makes no distinction between quotes:

```
public static boolean isQuote(char c) {
        switch (c) {
            case '\'':
            case '"':
            case '`':
                return true;
            default:
                return false;
        }
    }
```

it is a case of take it or leave it. :)

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|1|1|100.00%|

**Review by bziobrowski — APPROVED** [review]

**Review by ideoma — APPROVED** [review]
